terraform/modules/alb-listener-ip-restriction-rules/main.tf

```diff
@@ -0,0 +1,56 @@
+locals {
+  # Note: Needed due to an AWS quota restriction (limit of 5 conditions per rule)
+  # In future we may need to request a service quota increase of
+  # "Rules per Application Load Balancer" from AWS.
+  rules = setproduct(var.restricted_path_patterns, var.fully_trusted_source_ips)
+}
+
+# TODO: Uses count: This may lead to unintuitive behaviour when adding/removing
+# restricted_path_patterns and fully_trusted_source_ips. Impact could be
+# reduced by letting client assign priority to rules.
```

Comment:
My hunch is that we might need to do that in order to keep this maintainable in practice - I suspect this is gonna be a supercharged version of the problematic [lb-listener-rules](https://github.com/alphagov/govuk-aws/blob/master/terraform/modules/aws/lb_listener_rules/main.tf) and friends from govuk-aws (which became so brittle we can no longer add/remove things from it without major terraform-state surgery) but I might be wrong, so let's see how it goes.